src/miner.cpp

```diff
@@ -288,6 +290,45 @@ void BlockAssembler::SortForBlock(const CTxMemPool::setEntries& package, std::ve
     std::sort(sortedEntries.begin(), sortedEntries.end(), CompareTxIterByAncestorCount());
 }
 
+// This method is just a wrapper to addPackageTxs with all the setup required.
+CFeeRate BlockAssembler::minTxFeeRate()
+{
+    int nPackagesSelected = 0;
+    int nDescendantsUpdated = 0;
+    CFeeRate min_fee_rate = CFeeRate(COIN, 1);
+
+    resetBlock();
+    pblocktemplate.reset(new CBlockTemplate());
+
+    if(!pblocktemplate.get()){
+        return min_fee_rate;
+    }
+    pblock = &pblocktemplate->block; // pointer for convenience
+
+    // Add dummy coinbase tx as first transaction
+    pblock->vtx.emplace_back();
+    pblocktemplate->vTxFees.push_back(-1); // updated at end
+    pblocktemplate->vTxSigOpsCost.push_back(-1); // updated at end
+
+    CBlockIndex* pindexPrev = ::ChainActive().Tip();
+    assert(pindexPrev != nullptr);
+
+    pblock->nTime = GetAdjustedTime();
+    const int64_t nMedianTimePast = pindexPrev->GetMedianTimePast();
+
+    nLockTimeCutoff = (STANDARD_LOCKTIME_VERIFY_FLAGS & LOCKTIME_MEDIAN_TIME_PAST)
```

Comment:
IMO just set the nLocktimeCutoff to nMedianTimePast, as we care about up to date mempool all of them are far ahead BIP 113 activation.

Reply:
is there a downside to leaving as is? I'd rather have this code mimic the normal mining path as closely as possible.

Reply:
The downside is more complex code to review in the future.

Given current value `STANDARD_LOCKTIME_VERIFY_FLAGS`, `nLockTimeCutoff` is always going to be `nMedianTimePast`. And this rule being not only a standard one but a consensus one it's not going to change. We may have future different rules to compute transaction timelock but I think linking to standardness is a confusion here. Normal mining code should also do that.

Reply:
ok you've convinced me :) I'll update.

Reply:
fixed